README.md

```diff
@@ -232,23 +231,27 @@ Alternatively, you could use a repeatable movement managing plugin such as [nvim
 -- configuration
 require("nvim-treesitter-textobjects").setup {
     lsp_interop = {
-      border = 'none',
       floating_preview_opts = {},
     },
 }
 
 -- keymaps
 local peek_definition_code = require "nvim-treesitter-textobjects.lsp_interop".peek_definition_code
 -- You can use the capture groups defined in `textobjects.scm`
-vim.keymap.set({"n", "x"}, "<leader>df", peek_definition_code("@function.outer", "textobjects"))
-vim.keymap.set({"n", "x"}, "<leader>dF", peek_definition_code("@class.outer", "textobjects"))
+vim.keymap.set({"n", "x"}, "<leader>df", function()
+  peek_definition_code("@function.outer", "textobjects")
+end)
+vim.keymap.set({"n", "x"}, "<leader>dF", function()
+  peek_definition_code("@class.outer", "textobjects")
+end)
```

Comment:
Not quite clear what the point of this change is? This is just additional indirection without any benefit (since you `require` the module outside of the mapped function).

Reply:
I can move all the `require`s inside of the called functions. The change was mainly to avoid long lines.

Reply:
Yes, you _need_ to move the `require` inside to have any benefits. Otherwise you can just dispense with the `function() ... end` wrapper.

Reply:
Just to be clear and fully understand you. The purpose of the `function() ... end` wrapper is to invoke a lua function as callback of the keymap. I can't directly put `peek_definition_code("@class.outer", "textobjects")` because I would be putting an integer (this is what `peek_definition_code` returns) as a parameter of `vim.keymap.set`.

I'm interpreting your original message and your last message as if you thought that the purpose of the `function() ... end` wrapper was to lazily call the function. Is this correct? Maybe I'm just not understanding you right now. If that's the case, I'm sorry.

Originally, I thought that you didn't liked the indirection of requiring the function outside of the keymap and then calling it inside it, that's why I made the comment about being able to move all the `require`s inside the keymaps if that was preferred.

Reply:
Well, the function would return an integer either way? (The only difference is _when_ the integer is computed.)

But I trust you that the current code works; I'd still move the require inside so you only pay the cost when you're actually using a mapping; it's just good practice we should reinforce.